src/Microsoft.Health.Fhir.Core/Exceptions/PreconditionFailedException.cs

```diff
@@ -20,5 +21,19 @@ public PreconditionFailedException(string message)
                     OperationOutcomeConstants.IssueType.Required,
                     message));
         }
+
+        public PreconditionFailedException(string message, List<OperationOutcomeIssue> operationOutcomeIssues)
```

Comment:
Would it make more sense to just have something like a `TransactionFailedException` that takes in a status code and list of operation outcomes and does this logic? That way we're not repeating this in three different places since our main goal is to flow through the status code? Status code could even just default to `400`.

Inside of `OperationOutcomeExceptionFilterAttribute.cs` we could set the status code to the one that's set on the type.